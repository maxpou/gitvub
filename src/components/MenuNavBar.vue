<template>
  <header
    :class="[['Welcome'].indexOf($route.name) === -1 ? 'mdl-color--grey-800' : 'mdl-layout__header--transparent']"
    class="mdl-layout__header">
    <div class="mdl-layout__header-row">
      <router-link
        :to="{ name: 'Welcome' }"
        tag="span"
        class="mdl-layout-title"
      >GitVub</router-link>
      <div class="mdl-layout-spacer" />
      <div class="mdl-textfield mdl-js-textfield mdl-textfield--expandable mdl-textfield--floating-label mdl-textfield--align-right">
        <label
          class="mdl-button mdl-js-button mdl-button--icon"
          for="waterfall-exp">
          <i class="material-icons">search</i>
        </label>
        <div class="mdl-textfield__expandable-holder">
          <input
            id="waterfall-exp"
            v-model="userSearch"
            class="mdl-textfield__input"
            type="text"
            name="sample"
            @keyup.enter="proceedUserRepo()">
        </div>
      </div>
    </div>
  </header>
</template>

<script>
export default {
  name: 'MenuNavBar',
  data () {
    return {
      userSearch: ''
    }
  },
  computed: {
    canProceed () {
      return this.userSearch.length > 0
    }
  },
  methods: {
    proceedUserRepo () {
      if (this.canProceed) {
        this.$router.push({name: 'UserRepoList', params: { user: this.userSearch }})
        this.userSearch = ''
      }
    }
  }
}
</script>

<style scoped>
.mdl-layout-title {
  cursor: pointer;
  text-decoration: none;
}
</style>
