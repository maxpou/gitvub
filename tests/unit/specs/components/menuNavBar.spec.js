import { mount } from 'vue-test-utils'
import MenuNavBar from '@/components/MenuNavBar.vue'

describe('MenuNavBar.vue', () => {
  it('test initial rendering', () => {
    const wrapper = mount(MenuNavBar)
    const template = wrapper.html()
    expect(template).toMatchSnapshot()
  })
})
