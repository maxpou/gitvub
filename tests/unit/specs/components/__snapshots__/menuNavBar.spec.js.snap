// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuNavBar.vue test initial rendering 1`] = `"<nav class=\\"navbar navbar-expand-md navbar-dark fixed-top bg-dark\\"><router-link to=\\"[object Object]\\" class=\\"navbar-brand\\">GitVub</router-link></nav>"`;
